### PR TITLE
Initial draft of battery post

### PR DESCRIPTION
**Pull request by tyhoff**

### CODE REVIEW

File: _drafts/monitoring-battery-life.md [hunk 1]
```diff
@@ -0,0 +1,500 @@
...
+
+The general steps to accomplishing this are:
+
+- Ensure that your development boards have power rails so that you can isolate many of the components and you can easily attach probes to them.
+- Get a nice multimeter that can measure µA and mA.
+- Write a special firmware that instruments a single component in various power modes, and determine the current draw for each mode. This should be the simplest possible firmware and ideally only contain driver code.
+- Rinse and repeat with every component that could have a big impact on power consumption.
+
+For example, the accelerometer was a component in a wearable device that could consume a lot of power if left in the wrong state for periods of time. We wrote a firmware that would set it to different sampling rates and record the current consumption and then used this to determine at what sampling rate we could keep the accelerometer while still achieving our desired 7-day battery life.
```
Comment (bahildebrand) [inline _drafts/monitoring-battery-life.md:88]:
```suggestion
For example, the accelerometer is a component in a wearable device that could consume a lot of power if left in the wrong state for periods of time. We wrote firmware that would set it to different sampling rates and record the current consumption and then used this to determine at what sampling rate we could keep the accelerometer while still achieving our desired 7-day battery life.
```

Also the last sentence here runs on a bit. Consider restructuring.

File: _drafts/monitoring-battery-life.md [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+
+### State of Charge (SoC) over Time
+
+Once we determine whether or not our hardware can successfully meet our minimum requirements for battery life, now we need to put this to the test from the other direction. The primary thing we’ll look into now is how to measure the state of charge (SoC) of the battery over time to ensure that the device can last as long as required.
+
+One of the most common and simplest ways to measure the current capacity within a battery, or the SoC of a device, is by measuring the voltage of the battery during a known and consistent power draw, but it's not the only way.
+
+#### Coulomb Counting with Fuel Gauges
+
+At Pebble, we had a fuel gauge on one of the watches. A fuel gauge is a nifty hardware component that can indicate the battery’s SoC and health. It can understand the battery’s current capacity in two ways: measuring the voltage from the battery and learning the patterns over time, and also Coulomb counting, which measures how much current passes in and out of the battery over time.
```
Comment (bahildebrand) [inline _drafts/monitoring-battery-life.md:98]: Not sure if I'm reading this wrong, but isn't that three ways it measures batter capacity?

File: _drafts/monitoring-battery-life.md [hunk 3]
```diff
@@ -0,0 +1,500 @@
...
+  <img width="400" src="{% img_url monitoring-battery-life/battery-curve.png %}" alt="battery-curve" />
+</p>
+
+We need something better if possible.
+
+#### State of charge with voltage and battery curve
+
+One thing that can be done to help convert voltage to a percentage is to come up with a battery curve. A battery curve is simple: it's a map between a battery's voltage and the relative percentage that likely pertains to that voltage. Products also usually have both a charge and a discharge curve.
+
+A battery curve is what companies will have after they have some form of understanding of their battery properties and have spent some time. It is more easily understood by customer support teams and engineers that aren’t directly involved with the battery subsystem. That isn’t to say that it
```
Comment (bahildebrand) [inline _drafts/monitoring-battery-life.md:120]: `have spent some time`, not sure what should follow this. `have spent some time using the battery`?

**Review by bahildebrand — APPROVED** [review]
Awesome article!

File: _drafts/monitoring-battery-life.md [hunk 4]
```diff
@@ -0,0 +1,500 @@
+---
+title: Understanding Battery Performance of IoT Devices
+description: Monitoring & predicting the battery life of millions of embedded devices in production is made possible through the use of state of charge delta metrics and aggregations.
+author: tyler
```
[the remark this answers is not in the document]
Comment (tyhoff) [reply _drafts/monitoring-battery-life.md:4]: Oh! Forgot about them - they aren't in the `_template.md` so I forgot about them.

Will add some!

File: _drafts/monitoring-battery-life.md (hunk 1 above)
Comment (tyhoff) [reply _drafts/monitoring-battery-life.md:88]: Thanks. Will update.

File: _drafts/monitoring-battery-life.md (hunk 2 above)
Comment (tyhoff) [reply _drafts/monitoring-battery-life.md:98]: I'll make it cleaner

File: _drafts/monitoring-battery-life.md (hunk 3 above)
Comment (tyhoff) [reply _drafts/monitoring-battery-life.md:120]: Oof, fixed.

File: _drafts/monitoring-battery-life.md [hunk 5]
```diff
@@ -0,0 +1,500 @@
...
+
+A battery curve is what companies will have after they have some form of understanding of their battery properties and have spent some time. It is more easily understood by customer support teams and engineers that aren’t directly involved with the battery subsystem. That isn’t to say that it
+
+A nice tool that I came across this year at Embedded World was Qoitech[^qoitech], which builds a product to help users build charge and discharge curves[^qoitech_curves] under different environments. I believe their product is well worth the money if it can help companies translate a cryptic voltage reading to a percentage that everyone can understand.
+
+## Brief Primer on Metrics
+
+Before continuing to the following sections about capturing and aggregating metrics around battery life, let’s briefly discuss what a metric is, since I continue to come across firmware engineers who haven’t thought about them all that much.
+
+A **metric** is a measurement captured at runtime, and combining large numbers of metrics and calculating statistics is called an **aggregation**.
```
Comment (tyhoff) [inline _drafts/monitoring-battery-life.md:128]:
```suggestion
A **metric** is a measurement captured at runtime, and the process of combining large numbers of metrics and calculating statistics is known as **aggregation**.
```

File: _drafts/monitoring-battery-life.md [hunk 6]
```diff
@@ -0,0 +1,500 @@
...
+- camera on/off time
+- storage read/writes/erases
+
+### CPU & Code usage
+
+- CPU awake, sleep, and deep-sleep time
+- time spent running each task in the system
+- time running power-hungry blocks of code
+- boot time
+- number of logs written
```
[the remark this answers is not in the document]
Comment (tyhoff) [reply _drafts/monitoring-battery-life.md:189]: Ooh! Never thought about this one! Added.

**Pull request merged**